static/js/basic_async.js

```diff
@@ -0,0 +1,34 @@
+define([], function () {
+    function JsonGet(yourUrl) {
+        let req = new XMLHttpRequest(); // a new request
+        req.open("GET", yourUrl, false);
+        req.send(null);
+        return JSON.parse(req.responseText);
+    }
+
+    function asyncJsonGet(yourUrl, callback) {
+        if (!callback) {
+            throw "Callback needs to be a valid function";
+        }
+        let req = new XMLHttpRequest(); // a new request

```

Comment:
@SaltieRL/distributed-replays FYI, the JS `fetch` API has "good enough" browser support as long as IE is not cared about.

https://caniuse.com/#feat=fetch

86.67% global. I bet the audience for this would be 97%+. Meaning it will work directly in the browser with no transpiling, Babel or anything required.

`fetch` gives you a much clearer, native Promise based interface than raw XHR.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

This would be like:

```js
fetch(yourUrl)
  .then(function(res) { return res.json(); }) // next chained then will received the results of this, not original res
  .then(callback)
  .catch(console.error);
```

Reply:
I agree with this. For IE we can even use something like [this](https://github.com/github/fetch) as a polyfill.

Reply:
Sounds great!
This is why I tried to move it to its own file so that we can change how we do it.

Reply:
IMO, IE is dead. Like, what rocket league player is using IE 😆 

We would be in fairly good company if we chose not to support it and just redirecting people out to an upgrade page. TBH, often the IE traffic when I followed it down were bots anyway.

#LiveOnTheEdge (jk, but _edge_ does fully support fetch and most ES6 features natively and has surpassed IE globally leaving just computers who probably couldn't even render WebGL if you put an 1080TI in them)